File: ios/runanywhere-sdks-latest/examples/react-native/RunAnywhereAI/src/theme/system/dimens.ts
/**
 * Spacing / size / radius scale — ported from the Android example (ui/theme/Dimens.kt).
 * Read via `useTheme().dimens` so layout never hard-codes magic numbers.
 */
export const dimens = {
  screenPadding: 16,
  spacing: { xs: 4, sm: 8, md: 12, lg: 16, xl: 24 },
  icon: { sm: 18, md: 22, lg: 28 },
  radius: { sm: 8, md: 12, lg: 20, full: 999 },
  inputBarMinHeight: 48,
  contentMaxWidth: 760,
  navDrawerWidth: 260,
} as const;

export type Dimens = typeof dimens;

File: ios/runanywhere-sdks-latest/examples/android/RunAnywhereAI/app/src/main/java/com/runanywhere/runanywhereai/ui/theme/Dimens.kt
package com.runanywhere.runanywhereai.ui.theme

import androidx.compose.runtime.staticCompositionLocalOf
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp

// Spacing / size / radius scale. Read via LocalDimens.current — no magic numbers in UI code.
data class Dimens(
    val screenPadding: Dp,
    val spacingXs: Dp,
    val spacingSm: Dp,
    val spacingMd: Dp,
    val spacingLg: Dp,
    val spacingXl: Dp,
    val iconSm: Dp,
    val iconMd: Dp,
    val iconLg: Dp,
    val radiusSm: Dp,
    val radiusMd: Dp,
    val radiusLg: Dp,
    val radiusFull: Dp,
    val inputBarMinHeight: Dp,
    val contentMaxWidth: Dp,
    val navDrawerWidth: Dp,
    val bubbleMaxWidth: Dp,
)

val CompactDimens = Dimens(
    screenPadding = 16.dp,
    spacingXs = 4.dp,
    spacingSm = 8.dp,
    spacingMd = 12.dp,
    spacingLg = 16.dp,
    spacingXl = 24.dp,
    iconSm = 18.dp,
    iconMd = 22.dp,
    iconLg = 28.dp,
    radiusSm = 8.dp,
    radiusMd = 12.dp,
    radiusLg = 20.dp,
    radiusFull = 999.dp,
    inputBarMinHeight = 48.dp,
    contentMaxWidth = 760.dp,
    navDrawerWidth = 260.dp,
    bubbleMaxWidth = 320.dp,
)

val LocalDimens = staticCompositionLocalOf { CompactDimens }
